Enhance/display important dates on search

millerf opened the pull request:
* Finishing #41242  after a rebase

#### Changes proposed in this Pull Request

| Before | After | Comment|
| ---- | ---- | ----|
| ![Screenshot 2020-04-18 at 19 21 32](https://user-images.githubusercontent.com/790558/79644548-f3534c00-81a9-11ea-8f0b-182e5f3bfbe5.png) | ![Screenshot 2020-04-18 at 19 21 17](https://user-images.githubusercontent.com/790558/79644545-edf60180-81a9-11ea-9b9b-7be8ac429ee4.png) |  On listing, scheduled posts/pages now display the right date (scheduled ate) |
| ![Screenshot 2020-04-18 at 19 23 56](https://user-images.githubusercontent.com/790558/79644585-331a3380-81aa-11ea-8a6f-1e67a280ff57.png) |  ![Screenshot 2020-04-18 at 19 25 06](https://user-images.githubusercontent.com/790558/79644613-60ff7800-81aa-11ea-8b40-186fdbed6450.png) | Published now display "sticky" if needed | 
| ![Screenshot 2020-04-18 at 19 29 19](https://user-images.githubusercontent.com/790558/79644715-fef34280-81aa-11ea-9a0c-a3cd2b99fb72.png)  |  ![Screenshot 2020-04-18 at 19 26 25](https://user-images.githubusercontent.com/790558/79644651-91471680-81aa-11ea-97f4-afd2a801a14c.png)  | Post Search now displays the right status and appropriate date. The "sticky" status has been added. Maybe need some feedback from design? |
|  ![Screenshot 2020-04-18 at 19 29 05](https://user-images.githubusercontent.com/790558/79644751-45e13800-81ab-11ea-8dec-70fcf8510f54.png) | ![Screenshot 2020-04-18 at 19 31 03](https://user-images.githubusercontent.com/790558/79644737-2fd37780-81ab-11ea-813f-d060cce6b007.png)| Pages search now displays only one date and the right status |


Based on @sixhours remarks (https://github.com/Automattic/wp-calypso/pull/41242#pullrequestreview-396563709), some minor fixes need to be added.




Fixes #41075

a8ci18n commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,118 +20,139 @@ class PostRelativeTime extends React.PureComponent {
-		return time.fromNow();
+		// If the content is scheduled to be release within a year, do not display the year at the end
+		const scheduledTime = scheduledDate.calendar( null, {
+			sameElse: this.props.translate( '[on] ll [at] LT', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'LL [at] LT' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/pull/41301)

a8ci18n commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,118 +20,139 @@ class PostRelativeTime extends React.PureComponent {
-				statusText = this.props.translate( 'Publish immediately' );
-			}
+			const displayScheduleTime = this.getDisplayedTimeForLabel();
+			statusText = this.props.translate( 'scheduled %(displayScheduleTime)s', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **3** times:
`translate( 'scheduled for %(displayScheduleTime)s' )` *ES Score: 8*

millerf commented:
@sixhours 

>When searching for a scheduled post, the line "Scheduled for Tomorrow at [time]" reads oddly because "tomorrow" is capitalized:
![screenshot](https://user-images.githubusercontent.com/2124984/79775604-80111d80-8302-11ea-8b1f-d7f6188b0a1c.png)

That cannot be reproduced on this branch... It shows:
![Screenshot 2020-04-20 at 22 44 11](https://user-images.githubusercontent.com/790558/79798061-ae214c80-8358-11ea-805b-837bddbaeb23.png)
But I think the latter format was better, as it showed the hour.

sixhours commented:
I can't reproduce this behavior with this branch, either, but I can on production:

<img width="1083" alt="Screen Shot 2020-04-20 at 4 53 20 PM" src="https://user-images.githubusercontent.com/2124984/79798767-73eb8700-8327-11ea-85ec-58a5ee5af988.png">

So maybe we add the post time back, ie. "Scheduled in a day at [time]"?

millerf commented:
@sixhours I just pushed a fix.

a8ci18n commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,118 +20,143 @@ class PostRelativeTime extends React.PureComponent {
-		// Like "3 days ago" in English locale
-		return time.fromNow();
+		const scheduledTime = scheduledDate.calendar( null, {
+			nextDay: this.props.translate( '[tomorrow at] LT', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **38** times:
`translate( '[Tomorrow at] LT', { context: 'momentjs format string (nextDay)'} )` *ES Score: 12*

mmtr commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,118 +20,143 @@ class PostRelativeTime extends React.PureComponent {
+			nextDay: this.props.translate( '[tomorrow at] LT', {
+				comment: 'LT refers to time (eg. 18:00)',
+			} ),
+			sameElse: this.props.translate( '[on] ll [at] LT', {
```
I'm not a native English speaker, so I could be wrong, but it's "on" the right preposition to use after "scheduled"? Seems that can denote [when the post was scheduled rather than the actual schedule date](https://english.stackexchange.com/questions/72122/scheduled-on-vs-scheduled-for).

millerf commented:
> I'm not a native English speaker
Me neither... I'll change "on" to "for" right away...

a8ci18n commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,118 +20,150 @@ class PostRelativeTime extends React.PureComponent {
-
-		// Like "3 days ago" in English locale
-		return time.fromNow();
+		const sameElseForScheduled = this.props.translate( '[for] ll [at] LT', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'LL [at] LT' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/pull/41301)

a8ci18n commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,118 +20,150 @@ class PostRelativeTime extends React.PureComponent {
+				'll refers to date (eg. 21 Apr) for when the post will be published & LT refers to time (eg. 18:00) - "at" and "for" is translated',
+		} );
+
+		const sameElse = this.props.translate( '[on] ll [at] LT', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'LL [at] LT' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/pull/41301)

mmtr commented:
Thanks @millerf for taking this one!

Some quick notes after testing the posts search.

Before (prod) | After (this branch)
--- | ---
<img width="970" alt="Screen Shot 2020-04-21 at 11 42 35" src="https://user-images.githubusercontent.com/1233880/79851358-7b616d80-83c5-11ea-8b3b-a9290142c13a.png"> | <img width="974" alt="Screen Shot 2020-04-21 at 11 42 28" src="https://user-images.githubusercontent.com/1233880/79851401-8d431080-83c5-11ea-981e-5ee448056170.png">

- Not sure if the comma after "Draft" or "Published" is intended but I think we can drop it.
<img width="313" alt="Screen Shot 2020-04-21 at 11 48 23" src="https://user-images.githubusercontent.com/1233880/79851784-0e9aa300-83c6-11ea-8d04-332c65fdbf68.png">
<img width="230" alt="Screen Shot 2020-04-21 at 11 48 11" src="https://user-images.githubusercontent.com/1233880/79851786-0fcbd000-83c6-11ea-9759-fdc497ca2d82.png">

- Published posts should display the published date rather than the last modified date
<img width="327" alt="Screen Shot 2020-04-21 at 11 49 07" src="https://user-images.githubusercontent.com/1233880/79851856-2a9e4480-83c6-11ea-917c-3dbf145b6dd8.png">

- Scheduled posts for tomorrow should include the "for" preposition as well.
<img width="214" alt="Screen Shot 2020-04-21 at 11 49 57" src="https://user-images.githubusercontent.com/1233880/79851979-57eaf280-83c6-11ea-96c3-41a761eab7ec.png">

- Posts pending review should display the last modified date, along with the Pending review flag. I think we can consider them as drafts, since they have not been published yet.
<img width="266" alt="Screen Shot 2020-04-21 at 11 52 45" src="https://user-images.githubusercontent.com/1233880/79852891-9634e180-83c7-11ea-9c1d-33eef1e9b746.png">

millerf commented:
> Not sure if the comma after "Draft" or "Published" is intended but I think we can drop it.

It is intended, we discussed this [here](https://github.com/Automattic/wp-calypso/pull/41242#issuecomment-616680998). @sixhours thought it was misleading.

 > Scheduled posts for tomorrow should include the "for" preposition as well.

Will do

> Posts pending review 

How do you create those ones? 😅 
I remember we talked about it. I should be able blind-fix it anyway...

mmtr commented:
> It is intended, we discussed this [here](https://github.com/Automattic/wp-calypso/pull/41242#issuecomment-616680998). @sixhours thought it was misleading.

Uhm, I think she was referring specifically to the published posts combined with the last modified date. If we change that to read "Published on [published_date]" as I suggested above, I don't think we need the comma.

> How do you create those ones?

By toggling the "Pending review! checkbox on the sidebar panel:

<img width="352" alt="Screen Shot 2020-04-21 at 12 12 08" src="https://user-images.githubusercontent.com/1233880/79854252-78687c00-83c9-11ea-86d1-252eb4e9f517.png">

millerf commented:
@mmtr fixed. Sorry for the delay. I have been very busy lately...

a8ci18n commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,118 +20,160 @@ class PostRelativeTime extends React.PureComponent {
+		let scheduledTime;
+		if ( isScheduledPost ) {
+			scheduledTime = scheduledDate.calendar( null, {
+				nextDay: this.props.translate( '[for tomorrow at] LT', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **38** times:
`translate( '[Tomorrow at] LT', { context: 'momentjs format string (nextDay)'} )` *ES Score: 9*

a8ci18n commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,118 +20,160 @@ class PostRelativeTime extends React.PureComponent {
+				nextDay: this.props.translate( '[for tomorrow at] LT', {
+					comment: 'LT refers to time (eg. 18:00)',
+				} ),
+				sameElse: this.props.translate( '[for] ll [at] LT', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'LL [at] LT' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/pull/41301)

a8ci18n commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,118 +20,160 @@ class PostRelativeTime extends React.PureComponent {
-			// Like "Mar 15, 2013 6:23 PM" in English locale
-			return time.format( 'lll' );
+			scheduledTime = scheduledDate.calendar( null, {
+				sameElse: this.props.translate( '[on] ll [at] LT', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'LL [at] LT' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/pull/41301)

mmtr commented:
Thanks for the update @millerf and no need to apologize! You can take any time you need. Let me know what do you think about my notes below:

**Search: posts pending review**

<img width="250" alt="Screen Shot 2020-04-23 at 15 32 41" src="https://user-images.githubusercontent.com/1233880/80104645-c0c59c80-8577-11ea-82fc-8dfac96571ef.png">

Since posts pending reviews are actually drafts waiting for someone to review them, can we display the "Draft" status there and remove the modified date from the "pending review" flag? I think something like this would be ideal:

<img width="250" src="https://user-images.githubusercontent.com/1233880/80104828-02eede00-8578-11ea-9aed-4684a2fef988.png">

**Drafts list: posts pending review**

Also noted the pending review flag only appears when searching for posts. Can we display it as well when listing the drafts?

<img width="400" alt="Screen Shot 2020-04-23 at 15 38 42" src="https://user-images.githubusercontent.com/1233880/80105327-c079d100-8578-11ea-88ee-ebb922ab1f5c.png">

**Search: published posts**

There is an extra "on" we should remove.

<img width="250" alt="Screen Shot 2020-04-23 at 15 41 29" src="https://user-images.githubusercontent.com/1233880/80105477-f28b3300-8578-11ea-9ab1-3bf7b247d3b3.png">

**Published posts list**

Not sure we need to prefix the dates with "on". I think just the date reads ok, but don't have a strong opinion either so maybe both are fine. @Automattic/dotcom-manage-design, what's your feedback on this?

<img width="483" alt="Screen Shot 2020-04-23 at 15 42 29" src="https://user-images.githubusercontent.com/1233880/80105724-4007a000-8579-11ea-83b1-e31dffbe9484.png">

millerf commented:
@mmtr 

I modified "pending review" to be have a "draft" status, and added "pending" status as an "extra" sticker (the same way as "sticky" is).

I removed the duplicated "on"

I am waiting on the feedback of designers for the "On" on the listing...

And "future" was moved. Not sure why I left it there in the first place. I think it was to have an exhaustive list of status, even if "published" was still missing 🤷🏻‍♂️


| Search | Listing | 
| ---- | ---- | 
| <img width="620" alt="Screenshot 2020-04-24 at 19 42 49" src="https://user-images.githubusercontent.com/790558/80241552-0b244780-8664-11ea-8fe4-c30a2e4774da.png"> | <img width="497" alt="Screenshot 2020-04-24 at 19 42 40" src="https://user-images.githubusercontent.com/790558/80241540-06f82a00-8664-11ea-9203-64b7b867e6fa.png">|

sixhours commented:
I agree with @mmtr, the "On" isn't necessary. Just the date is fine. :)

jancavan commented:
> the "On" isn't necessary. Just the date is fine. :)

Somehow related, I don't think "tomorrow" needs to have the first letter capitalized?

Just quickly reading the comments (so I'm not sure if this has been addressed).

millerf commented:
> the "On" isn't necessary. Just the date is fine. :)

Done

> Somehow related, I don't think "tomorrow" needs to have the first letter capitalized?

That was already done.


I tried to address as much as I could. IT should be good now.

a8ci18n commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,118 +20,164 @@ class PostRelativeTime extends React.PureComponent {
+						'll refers to date (eg. 21 Apr) & LT refers to time (eg. 18:00) - "at" is translated',
+				} );
+			} else {
+				sameElseTranslation = this.props.translate( '[on] ll [at] LT', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'LL [at] LT' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/pull/41301)

mmtr commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,118 +20,164 @@ class PostRelativeTime extends React.PureComponent {
 	}
 
-	getDisplayedTimeFromPost( post ) {
+	getDisplayedTimeForLabel( noPrefix = false ) {
```
Optional: negated variables are usually hard to follow in code. Using `prefix = true` here will likely make the code a bit more easier to understand.

mmtr commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,118 +20,164 @@ class PostRelativeTime extends React.PureComponent {
+			statusClassName += ' is-trash';
+			statusIcon = 'trash';
+			const displayScheduleTime = this.getDisplayedTimeForLabel();
+			statusText = this.props.translate( 'trashed %(displayScheduleTime)s', {
```
@Automattic/i18n or @akirk can confirm, but I think we need to handle both status and date on the same translation. For instance, here `displayScheduleTime` contains `on [date]` for posts trashed more than a week ago, so `trashed %(displayScheduleTime)s` finally reads as `trashed on [date]`. Some languages might need to move the translated "on" part be before "trashed". To support those languages, the translatable string should be `trashed on [date]` rather than `trashed [datePrefixedWithOn]`.

The same applies for the rest of statuses (scheduled for [date], published on [date], draft last modified on [date], ...).

millerf replied:
This is quite complex to tackle, as the "on" is not always present (for example : "Published yesterday" vs "Published on Mar, 23 march 2020")
And as translation texts cannot have variables to pass on the status, I would have to have to list all cases that could happen.

Also waiting to translators to confirm, but I think the "on" is the beginning of the proposition of time, so it correlates more with the date than the adjective. As in "On 23rd of March, I published, etc,..."  But I might be mistaken...

mmtr commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,118 +20,164 @@ class PostRelativeTime extends React.PureComponent {
+					displayScheduleTime,
+				},
+			} );
+		} else if ( status === 'new' ) {
```
How can we test this status? Is it used for posts/pages that have been just published?

millerf replied:
I am not sure. I haven't been able to test it either. It is on master, though:

https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/post-relative-time-status/index.jsx#L133

mmtr commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,118 +20,164 @@ class PostRelativeTime extends React.PureComponent {
+	/**
+	 * Get status label
+	 *
+	 * @param {boolean} onlySticky sends back the "sticky" label. Special case as is using the same template but for is unrelated to the status
```
To avoid confusion, can we maybe separate concerns and create new functions for the sticky and pending review labels?

```
getStickyLabel() {
	return (
		<span className="post-relative-time-status__status is-sticky">
			<Gridicon icon="bookmark-outline" size={ 18 } />
			<span className="post-relative-time-status__status-text">{ this.props.translate( 'sticky' ) }</span>
		</span>
	);
}

getPendingReviewLabel() {
	return (
		<span className="post-relative-time-status__status is-pending">
			<Gridicon icon="aside" size={ 18 } />
			<span className="post-relative-time-status__status-text">{ this.props.translate( 'pending review' ) }</span>
		</span>
	);
}
```

That way we can get rid of the `onlySticky`/`onlyPending` arguments here, and used the specific functions for the labels [when needed](https://github.com/Automattic/wp-calypso/blob/00c312676cd04b2a33c26e80984ed03db27fe531/client/my-sites/post-relative-time-status/index.jsx#L199-L200): 

```
{ showPublishedStatus ? this.getStatusText() : timeText }
{ post.sticky && this.getStickyLabel() }
{ post.status === 'pending' && this.getPendingReviewLabel() }
```

millerf replied:
I didn't want to duplicate the HTML for maintenance purposes, but I think I found a good compromise:

https://github.com/Automattic/wp-calypso/pull/41301/files#diff-d7a7d45a786a1fd8012a5e9abc0a1d78R176-R200

mmtr commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,118 +20,164 @@ class PostRelativeTime extends React.PureComponent {
+				},
+			} );
+		} else if ( status === 'draft' || status === 'pending' ) {
+			const displayScheduleTime = this.getDisplayedTimeForLabel();
```
Non-blocking: Since this is not a scheduled post, we could drop any schedule reference here
```suggestion
			const displayTime = this.getDisplayedTimeForLabel();
```

The same applies for the rest of statuses.

mmtr commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,118 +20,164 @@ class PostRelativeTime extends React.PureComponent {
+			} );
+		} else if ( status === 'private' ) {
+			const displayScheduleTime = this.getDisplayedTimeForLabel();
+			statusText = this.props.translate( 'private last modified %(displayScheduleTime)s', {
```
Private posts should be considered as published as well. I noted we add a "Private" prefix to the title of these posts, so that should be enough: 

<img width="275" alt="Screen Shot 2020-04-27 at 11 55 26" src="https://user-images.githubusercontent.com/1233880/80359349-01762c00-887e-11ea-9942-5c37abeef556.png">

Ideally we could flag these posts using a label similar to the "Sticky"/"Pending reviews" ones and remove the title prefix, but it's fine to handle that in a follow-up.

millerf replied:
I did create an issue with that in mind a few days ago:
https://github.com/Automattic/wp-calypso/issues/41245

I'll reference it here to follow-up

mmtr commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,121 +20,184 @@ class PostRelativeTime extends React.PureComponent {
+			extraStatusClassName = 'is-trash';
+			statusIcon = 'trash';
+			const displayedTime = this.getDisplayedTimeForLabel();
+			statusText = this.props.translate( 'trashed %(displayedTime)s', {
```
Cannot add more comments to the [previous discussion](https://github.com/Automattic/wp-calypso/pull/41301#discussion_r415657120), so I'm copying the comments over here to continue it.

@mmtr https://github.com/Automattic/wp-calypso/pull/41301#discussion_r415657120:

> @Automattic/i18n or @akirk can confirm, but I think we need to handle both status and date on the same translation. For instance, here `displayScheduleTime` contains `on [date]` for posts trashed more than a week ago, so `trashed %(displayScheduleTime)s` finally reads as `trashed on [date]`. Some languages might need to move the translated "on" part before "trashed" (while leaving the date after). To support those languages, the translatable string should be `trashed on [date]` rather than `trashed [datePrefixedWithOn]`.
>
> The same applies for the rest of statuses (scheduled for [date], published on [date], draft last modified on [date], ...).

@millerf https://github.com/Automattic/wp-calypso/pull/41301#discussion_r415989089:

> This is quite complex to tackle, as the "on" is not always present (for example : "Published yesterday" vs "Published on Mar, 23 march 2020")
And as translation texts cannot have variables to pass on the status, I would have to have to list all cases that could happen.
> 
> Also waiting to translators to confirm, but I think the "on" is the beginning of the proposition of time, so it correlates more with the date than the adjective. As in "On 23rd of March, I published, etc,..."  But I might be mistaken...

mmtr replied:
> Also waiting to translators to confirm, but I think the "on" is the beginning of the proposition of time, so it correlates more with the date than the adjective

Yeah, let's see if @Automattic/i18n or @akirk can jump in. I might be wrong as well, but languages are so different that I wouldn't assume that prepositions are always part of the date in all of them.

mmtr commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,121 +20,184 @@ class PostRelativeTime extends React.PureComponent {
+	/**
+	 * Get Label for the status
+	 *
+	 * @param {*} statusText text status
```
Missed these jsdoc param types on my previous review. Is there any way we can make them more specific? I think they all are `string`, aren't they?

millerf replied:
They weren't there last time :) I had to refactor part of the component...

akirk commented:
I took a look at the screenshots and the code and it seems to me we still have a lot of string concats here which should be avoided. In some languages maybe the date needs to be put at the start of the phrase, this makes it impossible. Also it's hard to know if in a translation the date might depend on the rest of the phrase. Since there are not that many possible states, I'd recommend to just write them out:
- `Trashed on %(specificDate)s` (comment: Example: Trashed on March 3)
- `Trashed %(relativeTimeAgo)s` (comment: Example: Scheduled 3 days ago)
- `Published on %(specificDate)s` (comment: Example: Published on March 3)
- `Published %(relativeTimeAgo)s` (comment: Example: Published 3 days ago)
- `Last modified on %(specificDate)s` (comment: Example: Last modified on March 3)
- `Last modified %(relativeTimeAgo)s` (comment: Example: Last modified 3 days ago)
- `Scheduled %(futureRelativeDateAndTime)s` (comment: Example: Scheduled in 3 hours)
- `Scheduled for %(futureSpecificDateAndTime)s` (comment: Example: Scheduled for March 3)

Where possible, we should go for a label, like with `Pending Review`, we should do the same for `Private` and `Draft`. So for example:
- `Scheduled in 2 minutes <icon> Pending Review`
- `Scheduled for August 3 10:00 <icon> Private <icon> Sticky`
- `Last modified 3 days ago <icon> Draft <icon> Private`
- `Last modified on January 1 <icon> Draft <icon> Private`
- `Last modified on January 1 <icon> Pending Review`
- `Published 3 days ago <icon> Private <icon> Sticky`
- `Published on January 1 <icon> Private <icon> Sticky`
- `Trashed 3 days ago <icon> Private <icon> Sticky`
- `Trashed on January 1 <icon> Private <icon> Sticky`

Using a clock instead of "Last modified" and then the plain specific date or relative date is also an option. But it should only be used for "Last modified" and that date should only be displayed when the post is in the draft stage (resp. pending review).

mmtr commented:
> Where possible, we should go for a label, like with `Pending Review`, we should do the same for `Private` and `Draft`.

Not sure we need a `Draft` label. It is a status like trashed, published or scheduled. In fact, the last modified date will be only displayed on drafts, which is why we want to display posts in this state as `Draft last modified on [date]`/`Draft last modified [relativeTimeAgo]`.

millerf commented:
> Also it's hard to know if in a translation the date might depend on the rest of the phrase

Following our discussion about the right place of the `[on]`, please consider those examples:
```
Scheduled in 2 minutes
Scheduled for August 3 10:00
Published on January 1
```

`[in]`, `[for]` and `[on]` have the exact same syntaxic/grammatical function. So if we had to move [on] to the "published" translation block following your logic, we should also move [for] and [in] (although the last one is directly translated from the `moment` library, so it might be difficult).

I'll prepare the code for that anyway, but I think we really need some translator to help out here before doing anything 😄

akirk commented:
> [in], [for] and [on] have the exact same syntaxic/grammatical function.

In English. You cannot make this assumption for other languages, some languages don't have prepositions. I only know a few but I can well see cases where a relative time needs to come at the beginning of a sentence, while an absolute time needs to be at the end. This is why we should not split sentences and join them through concat.

a8ci18n commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,121 +20,200 @@ class PostRelativeTime extends React.PureComponent {
+		let displayedTime;
+		if ( isScheduledPost ) {
+			displayedTime = timestamp.calendar( null, {
+				nextDay: this.props.translate( '[tomorrow at] LT', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **38** times:
`translate( '[Tomorrow at] LT', { context: 'momentjs format string (nextDay)'} )` *ES Score: 12*

mmtr commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,121 +20,200 @@ class PostRelativeTime extends React.PureComponent {
+					} );
+				}
+				return this.props.translate( 'trashed %(displayedTime)s', {
+					comment: '%(displayedTime)s is when a post or page was trashed',
```
Given absolute/relative dates need to be at a different place for some languages, translators will need to know if this date is relative or absolute. Could you add some extra info in the comments so they can know that we use an absolute date in `trashed on/published on/scheduled for/last modified on [date]` and a relative date in `trashed/published/last modified [date]`?

millerf commented:
@mmtr Thanks a lot for the deep testing. Yesterday I didn't quite finish what I wanted to do, to it is still work-in-progress. Sorry I forgot to mention it!


But anyway, I found something interesting: 
> Private posts are flagged twice now: with a prefix in the title and with a status label. Fine to address this in a follow-up.

I didn't have any private post, so I didn't see it.
It turns out it is coming from the back-end. They prefix the post title with "private: " when searching.

```
Request URL: https://public-api.wordpress.com/rest/v1.1/sites/174604573/posts?http_envelope=1&author=&number=20&order=DESC&search=&status=publish%2Cprivate&type=post&page=1
``` 
<img width="342" alt="Screenshot 2020-04-29 at 18 03 44" src="https://user-images.githubusercontent.com/790558/80618659-f3720800-8a43-11ea-9880-f715abf3b570.png">

millerf commented:
>Published date is not displayed for private posts. We'd need to fix this before merging.

Fixed

millerf commented:
I found another bug on `wordpress.com` that I will fix now:
You can click on a trashed page, and it sends you to a `404`...

kwight commented:
> You can click on a trashed page, and it sends you to a 404... Posts cannot be clicked when trashed.

Well spotted @millerf , definitely a bug: https://github.com/Automattic/wp-calypso/issues/41640

mmtr commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,121 +20,208 @@ class PostRelativeTime extends React.PureComponent {
+					<>
+						<Gridicon icon="time" size={ this.props.gridiconSize || 18 } />
+						<time className="post-relative-time-status__time-text" dateTime={ time }>
+							{ this.getDisplayedTimeForLabel( true ) }
```
`getDisplayedTimeForLabel` no longer needs arguments.
```suggestion
							{ this.getDisplayedTimeForLabel() }
```

mmtr commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,121 +20,208 @@ class PostRelativeTime extends React.PureComponent {
+				} ),
+				sameElse: this.props.translate( 'll [at] LT', {
+					comment:
+						'll refers to date (eg. 21 Apr) for when the post will be published & LT refers to time (eg. 18:00) - and "for" is translated',
```
Seems this refers to an old message.
```suggestion
						'll refers to date (eg. 21 Apr) for when the post will be published & LT refers to time (eg. 18:00) - "at" is translated',
```

mmtr commented:
> It turns out it is coming from the back-end. They prefix the post title with "private: " when searching.

Nice catch! I'll file an issue so we can fix that server-side once this lands.

millerf commented:
@mmtr I commited your suggestions, I think this can be landed safely now
